x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityImplicitBehaviorBootstrapCheckTests.java

```diff
@@ -74,7 +88,8 @@ public void testUpgradeFrom7xWithExplicitSecuritySettings() throws Exception {
         );
         NodeMetadata nodeMetadata = new NodeMetadata(randomAlphaOfLength(10), previousVersion, Version.CURRENT);
         nodeMetadata = nodeMetadata.upgradeToCurrentVersion();
-        BootstrapCheck.BootstrapCheckResult result = new SecurityImplicitBehaviorBootstrapCheck(nodeMetadata).check(
+        LicenseService licenseService = mock(LicenseService.class);
```

Comment:
Mocking just the interface essentially bypasses this test because it will fail the `instanceof ClusterStateLicenseService` check and return immediately with success. We can add a separate test method for this short-circuiting logic. But it is *not* the intention of this test method. Here we should still mock `ClusterStateLicenseService` so that we test the code block inside the `if (... instanceof ...)` branch.

Reply:
good catch, thanks !